### DISCUSSION TITLE
"Request for Zoom Feature in NetBeans IDE for Improved Visibility"

### DISCUSSION THREAD

**Discussion by mustafa9514** [apache/netbeans#6676] [Ideas]
"I noticed that NetBeans IDE currently lacks a zoom feature for the view settings, and I would like to suggest its addition. The windows and components appear quite small, which strains my eyes during usage. Adding a zoom feature would greatly improve visibility and reduce eye strain. I believe this enhancement would contribute to a more user-friendly experience. Thank you for considering this suggestion, as it would make a significant difference for users who may encounter similar challenges. Your attention to this matter is greatly appreciated."

**Comment by mustafa9514** [comment]
"Subject: Adding a Zoom Feature for Enhanced User Comfort in NetBeans IDE

Dear NetBeans Team,

I hope this message finds you well. I am a regular user of NetBeans IDE and have a suggestion that I believe would significantly improve the user experience.

I've noticed that the current lack of a zoom feature in the view settings is causing some discomfort for users like myself. The windows and components appear quite small, leading to eye strain during prolonged usage. To address this, I kindly request the addition of a zoom feature that allows users to adjust the screen size as needed.

The ability to zoom in and out would not only enhance visibility but also contribute to a more comfortable working environment. This feature would be especially beneficial for users who spend extended periods using NetBeans.

I appreciate your attention to user feedback and believe that incorporating this zoom feature would greatly contribute to the overall satisfaction of NetBeans users.

Thank you for your time and consideration.

Best regards,

**Reply by neilcsmith-net** [reply]
Note that you can zoom the source editor using `ALT+MOUSE_WHEEL`.  I realise this doesn't cover everything you've mentioned.

You can also run NetBeans with eg. the `--fontsize 16` option, or with `-J-Dflatlaf.uiScale=1.5`.  That requires editing `netbeans.conf` or the command line in the windows shortcut, desktop entry, etc.

You could also try using `Tools / Options / Appearance / FlatLaf : Edit custom properties` and add the line `defaultFont = +6`, save the file and restart NetBeans.  The latter might be a good option in the main FlatLaf settings.

**Reply by mustafa9514** [reply]
You're welcome! I'm glad it worked. The entire screen enlarged, and my eyes feel more comfortable.

**Reply by goyalyashpal** [reply]
hey @mustafa9514 - please mark as answer and close the discussion :)
